privly.safariextension/scripts/privly.js

```diff
@@ -3,269 +3,904 @@ Open Source Initiative OSI - The MIT License (MIT):Licensing
-    iFrame.setAttribute("id","ifrm"+privly.nextAvailableFrameID);
-    iFrame.setAttribute("frameborder","0");
-    privly.nextAvailableFrameID++;
-    iFrame.setAttribute("style","width: 100%; height: 32px; overflow: hidden;");
-    iFrame.setAttribute("scrolling","no");
-    iFrame.setAttribute("overflow","hidden");
-    
-    object.parentNode.replaceChild(iFrame, object);
+    var attrs= {
+      "frameborder":"0",
+      "vspace":"0",
+      "hspace":"0",
+      "width":"100%",
+      "marginwidth":"0",
+      "marginheight":"0",
+      "height":"1px",
+      "style":"width: 100%; height: 32px; " + "overflow: hidden;",
+      "scrolling":"no",
+      "overflow":"hidden",
+      "data-privly-display":"true",
+      "data-privly-accept-resize":"true", //Custom attribute indicating this iframe
+      //is eligible for being resized by its contents
+      "src":applicationUrl,
+      "id":"ifrm" + id, //The id and the name are the same so that the iframe can be
+      "name":"ifrm" + id //uniquely identified and resized
+       };
+
+    //Styling and display attributes
+     for(var key in attrs) 
+     {
+       iFrame.setAttribute(key, attrs[key]);
+     }
+
+    //Determines whether the element will be shown after it is toggled.
+    //This allows for the button to turn on and off the display of the
+    //injected content.
+    if ( object.getAttribute("data-privly-display") !== "off" ) {
+      object.setAttribute("data-privly-display", "false");
+    }
+    object.style.display = "none";
+
+    //put the iframe into the page
+    object.parentNode.insertBefore(iFrame, object);
+  },
+
+  /**
+   * Replace an anchor element with its referenced content. This function
+   * will opt for a locally stored application if there is one, otherwise
+   * it will inject the remote code.
+   *
+   * @param {object} object A hyperlink element to be replaced
+   * with an iframe referencing its content
+   */
+  injectLink: function(object)
+  {
+    "use strict";
+
+    // Sets content URL.
+    var frameId = privly.nextAvailableFrameID++;
+    var iframeUrl = object.getAttribute("data-privlyHref");
+
+    // Only the Chrome extension currently supports local code storage.
+    // other extensions will default to remote code execution.
+    if (typeof chrome !== "undefined" && chrome.extension !== undefined &&
+      chrome.extension.sendMessage !== undefined) {
+        chrome.extension.sendMessage(
+          {privlyOriginalURL: iframeUrl},
+          function(response) {
+            if( typeof response.privlyApplicationURL === "string" ) {
+              privly.injectLinkApplication(object, response.privlyApplicationURL, frameId);
+            }
+          });
+      } else if (typeof safari !== "undefined" && safari.extension !== undefined) {
+        var path;
+        if( iframeUrl.indexOf("privlyApp=Message") > 0 || iframeUrl.indexOf("privlyApp=ZeroBin") > 0 ) {
+          path = "privly-applications/Message/show.html?privlyOriginalURL=";
```

Comment:
You should not hard code the application path. You should be able to do something like `path = "privly-applications/Message/" + appName + ".html?privlyOriginalURL=";`